Redis constructor type does not fully express its valid uses

[`@keyv/redis`'s constructor signature](https://github.com/jaredwray/keyv/blob/main/packages/redis/src/index.d.ts#L7) does not support passing it a ` Redis.Cluster` or `Redis` instance which looks like it's supported based on the [implementation](https://github.com/jaredwray/keyv/blob/main/packages/redis/src/index.js#L11)

`@keyv/redis` version `2.3.1`

@jaredwray the types don't seem quite. Digging into it.

With `import {ClientOpts} from 'redis'` will result in 
```
Cannot find module 'redis' or its corresponding type declarations.ts(2307)
```
since `redis` is not dependency or peer dependency of this package. 

Which means the types for this use case don't match up
```ts
const KeyvRedis = require('@keyv/redis');
const Redis = require('ioredis');
const Keyv = require('keyv');

const redis = new Redis('redis://user:pass@localhost:6379');
const keyvRedis = new KeyvRedis(redis);
const keyv = new Keyv({ store: keyvRedis });
```

I wonder if `import ClientOpts from 'ioredis';` was meant which I think would work for `uri instanceof Redis` use case but not for `uri instanceof Redis.Cluster`

Would it make sense to just do
```ts
import {Redis, Cluster} from 'ioredis';

declare class KeyvRedis extends EventEmitter implements Store<string | undefined> {
	readonly ttlSupport: false;
	namespace?: string | undefined;
	opts: Record<string, unknown>;
	redis: any;
	constructor(options?: KeyvRedis.Options | Redis | Cluster);
	constructor(uri: string, options?: KeyvRedis.Options);
	get(key: string): Promise<string | undefined>;
	getMany(keys: string[]): Promise<string[] | undefined>;
	set(key: string, value: string | undefined, ttl?: number): Promise<any>;
	delete(key: string): boolean;
	deleteMany(keys: string[]): boolean;
	clear(): Promise<void>;
	iterator(namespace: string | undefined): AsyncGenerator<any, void, any>;
	has(key: string): boolean;
}
declare namespace KeyvRedis {
	interface Options {
		uri?: string | undefined;
		dialect?: string | undefined;
	}
}
```
Instead

--- a/packages/redis/src/index.d.ts
+++ b/packages/redis/src/index.d.ts
@@ -1,14 +1,14 @@
 import {EventEmitter} from 'events';
 import {Store} from 'keyv';
-import {ClientOpts} from 'redis';
+import {Redis, Cluster} from 'ioredis';
 
 declare class KeyvRedis extends EventEmitter implements Store<string | undefined> {
 	readonly ttlSupport: false;
 	namespace?: string | undefined;
 	opts: Record<string, unknown>;
 	redis: any;
-	constructor(options?: KeyvRedis.Options);
-	constructor(uri: string, options?: KeyvRedis.Options);
+	constructor(options?: KeyvRedis.Options | Redis | Cluster);
+	constructor(uri: string | Redis | Cluster, options?: KeyvRedis.Options);
 	get(key: string): Promise<string | undefined>;
 	getMany(keys: string[]): Promise<string[] | undefined>;
 	set(key: string, value: string | undefined, ttl?: number): Promise<any>;
@@ -19,7 +19,7 @@ declare class KeyvRedis extends EventEmitter implements Store<string | undefined
 	has(key: string): boolean;
 }
 declare namespace KeyvRedis {
-	interface Options extends ClientOpts {
+	interface Options {
 		uri?: string | undefined;
 		dialect?: string | undefined;
 	}

--- a/packages/redis/src/index.d.ts
+++ b/packages/redis/src/index.d.ts
@@ -1,14 +1,17 @@
 import {EventEmitter} from 'events';
+import { Store } from 'keyv';
+import { ClientOpts } from 'redis';
 
-declare class KeyvRedis extends EventEmitter {
+declare class KeyvRedis extends EventEmitter implements Store<string | undefined> {
 	readonly ttlSupport: false;
+	namespace?: string | undefined;
 	opts: Record<string, unknown>;
 	redis: any;
-	constructor(options?: string | KeyvRedis.Options);
-	_getNamespace(): string;
+	constructor(options?: KeyvRedis.Options);
+	constructor(uri: string, options?: KeyvRedis.Options);
 	get(key: string): Promise<string | undefined>;
 	getMany(keys: string[]): Promise<string[] | undefined>;
-	set(key: string, value: string | undefined): Promise<any>;
+	set(key: string, value: string | undefined, ttl?: number): Promise<any>;
 	delete(key: string): boolean;
 	deleteMany(keys: string[]): boolean;
 	clear(): Promise<void>;
@@ -16,7 +19,7 @@ declare class KeyvRedis extends EventEmitter {
 	has(key: string): boolean;
 }
 declare namespace KeyvRedis {
-	interface Options {
+	interface Options extends ClientOpts {
 		uri?: string | undefined;
 		dialect?: string | undefined;
 	}

--- a/packages/redis/src/index.d.ts
+++ b/packages/redis/src/index.d.ts
@@ -1,6 +1,6 @@
 import {EventEmitter} from 'events';
-import { Store } from 'keyv';
-import { ClientOpts } from 'redis';
+import {Store} from 'keyv';
+import {ClientOpts} from 'redis';
 
 declare class KeyvRedis extends EventEmitter implements Store<string | undefined> {
 	readonly ttlSupport: false;

